src/Symfony/Component/Mailer/SmtpEnvelope.php

```diff
@@ -39,7 +39,7 @@ public static function create(RawMessage $message): self
 
     public function setSender(Address $sender): void
     {
-        $this->sender = new Address($sender->getAddress());
+        $this->sender = $sender;
```

Comment:
This was done this way as the Sender (`MAIL FROM` in the SMTP protocol) only accepts an address without the "name" part. But indeed, now that we are also using it for HTTP/API, we need to keep the full address. You must fix `$this->doMailFromCommand($envelope->getSender()->toString());` in `SmtpTransport` though to only use the address.